finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Row.scala

```diff
@@ -71,14 +71,15 @@ class StringEncodedRow(rawRow: Buf, val fields: IndexedSeq[Field], indexMap: Map
       else {
         val str = new String(bytes, Charset(charset))
         field.fieldType match {
-          case Type.Tiny       => ByteValue(str.toByte)
-          case Type.Short      => ShortValue(str.toShort)
-          case Type.Int24      => IntValue(str.toInt)
-          case Type.Long       => IntValue(str.toInt)
-          case Type.LongLong   => LongValue(str.toLong)
-          case Type.Float      => FloatValue(str.toFloat)
-          case Type.Double     => DoubleValue(str.toDouble)
-          case Type.Year       => ShortValue(str.toShort)
+          case Type.Tiny                        => ByteValue(str.toByte)
+          case Type.Short                       => ShortValue(str.toShort)
+          case Type.Int24                       => IntValue(str.toInt)
+          case Type.Long if field.isUnsigned()  => LongValue(str.toLong)
```

Comment:
The thing is in order to store 9223372036854775808 in a column of MySQL Table, that column should be bigint. If the field is bigint then its type is Type.LongLong. Then, I guess adding another value type, let's say, BigIntValue and another function to MysqlBufReader, like **readUnsignedLong** which is returning a BigInt should solve our problem.